src/domainPaths/domainPath.ts

```diff
@@ -236,13 +237,12 @@ export class DomainPath {
    *
    *
    * @param page - The puppeteer page to which we can attach listeners or change behaviour of
-   * @param bot - Optional. The {@link Bot} to use for authentication.
+   * @param selectedBot - Optional. The {@link Bot} to use for authentication.
    * @param config - Optional. The {@link BrowserConfig} passed by the user
    */
   // eslint-disable-next-line class-methods-use-this
-  public async pageHandler (page: PuppeteerPage, selectedBot?: Bot, config?: any): Promise<IAuthlessResponse | null> {
-    // default implementation to process the page
-    return null
+  public async pageHandler (page: PuppeteerPage, selectedBot?: Bot, config?: any): Promise<IAuthlessResponse> {
```

Comment:
as mentioned, `pageHandler` should return an `AuthlessResponse` instance. I don't care if the returned object has the same shape as `IAuthlessResponse`. I want an actual `AuthlessResponse` instance so I can call `toResources` etc. on it.

```suggestion
  public async pageHandler (page: PuppeteerPage, selectedBot?: Bot, config?: any): Promise<AuthlessResponse> {
```